levels.go

```diff
@@ -351,7 +352,10 @@ func (s *levelsController) startCompact(lc *z.Closer) {
 	n := s.kv.opt.NumCompactors
 	lc.AddRunning(n - 1)
 	for i := 0; i < n; i++ {
-		go s.runCompactor(i, lc)
+		i := i
+		s.kv._go(func() {
+			s.runCompactor(i, lc)
+		})
```

Comment:
`s.kv` is an instance of DB, so I assumed it was safe to start the Goroutine using this instance. But I am not 100% sure this instance is directly derived from the DB instance the user is using, so tell me if this is a mistake.